### ISSUE TITLE
useMemo returns `undefined` even when it should never do it (as of 10.8.1)

### ISSUE THREAD

**Issue by rburgst** [preactjs/preact#3582]
- [x] Check if updating to the latest Preact version resolves the issue

**Describe the bug**

The following code produces an error on the 1st re-render

```typescript
export const TestComponent: FunctionComponent = () => {
    const ref = useRef(null)
    const [currentBreakpoints, setCurrentBreakpoints] = useState<number[]>([]);    

    const result = useMemo(
        () => currentBreakpoints.map((breakpoint: number) => `cq-${breakpoint}`).join(' '),
        [currentBreakpoints]
    );
    if (result === undefined) {
        throw new Error('hooks are broken!');
    }
    
    return <div class={result} ref={ref}></div>
}
```

**To Reproduce**

check out the repo at https://github.com/rburgst/preact-error-usememo

then simply do 

1. `yarn install`
2. `yarn jest`

with preact < 10.8.1 everything works as expected, with preact `10.8.1` the re-render produces `result === undefined`.

**Expected behavior**
The test should pass on 10.8.1.

**Comment by JoviDeCroock** [comment]
I think this should be fixed by the latest few PR's by @marvinhagemeister

**Comment by rburgstaller** [comment]
this appears to be fixing the bug: https://github.com/preactjs/preact/pull/3580

**Comment by JoviDeCroock** [comment]
I will close this as it's solved in that PR and we will release 10.8.2 asap

**Comment by rburgst** [comment]
I can confirm that 10.8.2 fixes the issue
